python/sqlalchemy/performance-improvements.yaml

```diff
@@ -0,0 +1,13 @@
+rules:
+  - id: len-all-count
+    pattern: len($X.all())
+    message: "Using QUERY.count() instead of len(QUERY.all()) sends less data to the client since the SQLAlchemy method is performed server-side."
+    languages: [python]
+    severity: WARNING
+  - id: batch-import
+    pattern: |
+      for $X in $Y:
+        db.session.add($Z)
+    message: "Rather than adding one element at a time, consider batch loading to improve performance."
```

Comment:
Which function does batch loading?

Reply:
Adding multiple at once from [this article](https://dev.to/zchtodd/sqlalchemy-performance-anti-patterns-and-their-fixes-4bmm). 

```
batch = []
insert_stmt = Object.__table__.insert()
for $X in $Y:
    if len(batch) > 1000:
       db.session.execute(insert_stmt, batch)
       batch.clear()
    batch.append($X)
if batch:
    db.session.execute(insert_stmt, batch)
```